data-cleaning/notebook.ipynb

```diff
@@ -286,13 +314,187 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 12,
+   "execution_count": 10,
    "id": "47fef19d",
    "metadata": {},
    "outputs": [],
    "source": [
-    "df.to_json('./data.json', force_ascii=False, orient='table', index=False)"
+    "#df.to_json('./data.json', force_ascii=False, orient='table', index=False)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f3ae4c42",
+   "metadata": {},
+   "source": [
+    "# Counting null values - Deleting videos with empty titles - Removing repeated tag"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "4af22e79",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "url            0\n",
+      "title          0\n",
+      "description    0\n",
+      "tags           0\n",
+      "thumbnail      0\n",
+      "dtype: int64\n",
+      "shape before drop: (12610, 5)\n",
+      "indexes with empty titles: [706, 731, 786, 7889, 12179, 12201, 12228, 12253, 12301]\n",
+      "original shape: (12849, 5) updated shape: (12601, 5)\n",
+      "check if there is empty titles: 0\n"
+     ]
+    }
+   ],
+   "source": [
+    "#pd.set_option('display.max_rows', None)\n",
+    "\n",
+    "# create a copy of last version of complete_dataframe\n",
+    "update_df = df.copy()\n",
+    "\n",
+    "# find indexes where there is just an empty string\n",
+    "indexes = update_df[update_df['title'] == ''].index.tolist()\n",
+    "# Counting null data in each column\n",
+    "print(update_df.isnull().sum())\n",
+    "\n",
+    "print(\"shape before drop:\", update_df.shape)\n",
+    "print(\"indexes with empty titles:\", indexes)\n",
+    "update_df = update_df.drop(index = indexes)\n",
+    "print(\"original shape:\", complete_dataframe.shape, \"updated shape:\", update_df.shape)\n",
+    "\n",
+    "#checking if there is another empty title in the updated\n",
+    "print(\"check if there is empty titles:\", len(update_df[update_df['title'] == ''].index))\n",
+    "\n",
+    "# reseting the index\n",
+    "update_df = update_df.reset_index(drop=True)\n",
+    "\n",
+    "# remove tag that contains: 'video, compartir, teléfono con cámara, teléfono con video, gratuito, subir'\n",
+    "update_df.loc[update_df['tags'] == 'video, compartir, teléfono con cámara, teléfono con video, gratuito, subir', 'tags'] = ''"
```

Comment:
¿Cuántos vídeos se afectan con esta operación? Según veo se eliminan todos los tags de esos vídeos, ¿Solo tienen ese tag?

Reply:
La cantidad exacta, no la conté en esta ocasión, pero cuando lo hice, eran alrededor de unos 1970 o 1978 no lo recuerdo bien.
Sólo se afectan los vídeos que tengan en su columna de 'tags' un valor igual a: 'video, compartir, teléfono con cámara, teléfono con video, gratuito, subir'. Sólo tienen ese tag, si tuviesen alguno otro, según entiendo ya la condición no se cumpliría. Si algo, podrías ayudarme a corroborarlo, para ello me valí del json que se crea antes de esta ejecución y el posterior, y los comparé, y vi que ocurre el cambio de manera satisfactoria (para ello, creo que se debe descomentar uno de los data.json que está en medio). Aquí hay un poco de documentación con ejemplos, sobre lo que hace .loc en pandas: https://www.analyticslane.com/2019/06/21/seleccionar-filas-y-columnas-en-pandas-con-iloc-y-loc/

Reply:
Confirmo, en los datos iniciales (sin ningún tratamiento) encuentro 2030 coincidencias, son vídeos que solo tienen ese tag. Quería confirmar que no era un `.contains` que eliminaba todos los tags de un vídeo que tuviera el tag de la condición.

Reply:
Gracias por confirmar! Mala mía poner contains en el comentario, debí decir, equals to... O algo así. Lo tendré presente.